ghost/members-api/test/unit/lib/controllers/router.test.js

```diff
@@ -388,5 +388,52 @@ describe('RouterController', function () {
                 await controller.sendMagicLink(req, res).should.be.rejectedWith(`Cannot subscribe to archived newsletters Newsletter 2`);
             });
         });
+
+        describe('honeypot', function () {
+            let req, res, sendEmailWithMagicLinkStub;
+
+            const createRouterController = (deps = {}) => {
+                return new RouterController({
+                    allowSelfSignup: sinon.stub().returns(true),
+                    memberAttributionService: {
+                        getAttribution: sinon.stub().resolves({})
+                    },
+                    sendEmailWithMagicLink: sendEmailWithMagicLinkStub,
+                    ...deps
+                });
+            };
+
+            beforeEach(function () {
+                req = {
+                    body: {
+                        email: 'jamie@example.com',
+                        emailType: 'signup'
+                    },
+                    get: sinon.stub()
+                };
+                res = {
+                    writeHead: sinon.stub(),
+
+                    end: sinon.stub()
+                };
+                sendEmailWithMagicLinkStub = sinon.stub().resolves();
+            });
+
+            it('Sends emails when honeypot is not filled', async function () {
+                const controller = createRouterController();
+
+                req.body.honeypot = 'filled!';
+
+                await controller.sendMagicLink(req, res).should.be.fulfilled();
+                sendEmailWithMagicLinkStub.notCalled.should.be.true();
```

Comment:
Isn't this incorrect? Shouldn't we see the email stub called?

The inverse being true for the following test.

Reply:
The honeypot is a fake input for a mobile phone number. The idea is that if a bot fills it then we _shouldn't_ send the email, since they've fallen into the trap :-)

Reply:
Yes but I think you have the titles for the tests backwards, then? Either the content or test titles/descriptions need to swap. Functionally I think you're all good 👍.